### DISCUSSION TITLE
Is there a trick in order to send and recieve unencrypted data?

### DISCUSSION THREAD

**Discussion by mpsOxygen** [PiotrMachowski/Xiaomi-cloud-tokens-extractor#30] [Q&A]
I was playing around with intercepting the data that the Mi Home app sends to the cloud and I noticed that the data field looks encrypted. From what I can see in the header ( Miot-Encrypt-Algorithm: ENCRYPT-RC4 ) they use RC4. Is there a header I can change in order to force the exchange to be plain text?

**Comment by PiotrMachowski** [comment]
I am not able to answer your question, I based this work on existing projects :( 
You can check out [modded Mi Home app](https://www.kapiba.ru/), maybe this will help you

**Reply by mpsOxygen** [reply]
No problem, I already found how. It's here in case anyone else needs it: https://github.com/rytilahti/python-miio/issues/185#issuecomment-778703191
